napari/layers/base/_base_mouse_bindings.py

```diff
@@ -0,0 +1,171 @@
+import warnings
+
+import numpy as np
+
+from napari.layers.utils.interaction_box import (
+    InteractionBoxHandle,
+    generate_transform_box_from_layer,
+    get_nearby_handle,
+)
+from napari.utils.transforms import Affine
+
+
+def highlight_box_handles(layer, event):
+    if not len(event.dims_displayed) == 2:
+        return
+
+    # we work in data space so we're axis aligned which simplifies calculation
+    # same as Layer.world_to_data
+    world_to_data = (
+        layer._transforms[1:].set_slice(event.dims_displayed).inverse
+    )
+    pos = np.array(world_to_data(event.position))[event.dims_displayed]
+
+    handle_coords = generate_transform_box_from_layer(
+        layer, event.dims_displayed
+    )
+    # TODO: dynamically set tolerance based on canvas size so it's not hard to pick small layer
+    nearby_handle = get_nearby_handle(pos, handle_coords)
+
+    # set the selected vertex of the box to the nearby_handle (can also be INSIDE or None)
+    layer._overlays['transform_box'].selected_vertex = nearby_handle
+
+
+def transform_with_box(layer, event):
+    if not len(event.dims_displayed) == 2:
+        return
+
+    # we work in data space so we're axis aligned which simplifies calculation
+    # same as Layer.data_to_world
+    data_to_world = layer._transforms[1:].simplified.set_slice(
+        event.dims_displayed
+    )
+    world_to_data = data_to_world.inverse
+    pos = np.array(event.position)[event.dims_displayed]
+    pos_data = world_to_data(pos)
+
+    initial_handle_coords_data = generate_transform_box_from_layer(
+        layer, event.dims_displayed
+    )
+    nearby_handle = get_nearby_handle(pos_data, initial_handle_coords_data)
+
+    if nearby_handle is None:
+        return
+
+    # now that we have the nearby handles, other calculations need
+    # the world space handle positions
+    initial_handle_coords = data_to_world(initial_handle_coords_data)
+
+    # initial layer transform so we can calculate changes later
+    initial_affine = layer.affine.set_slice(event.dims_displayed)
+    initial_data2physical = layer._transforms['data2physical'].set_slice(
+        event.dims_displayed
+    )
+    initial_position = pos
+
+    # needed for resize and rotate
+    center = np.mean(
+        initial_handle_coords[
+            [
+                InteractionBoxHandle.TOP_LEFT,
+                InteractionBoxHandle.BOTTOM_RIGHT,
+            ]
+        ],
+        axis=0,
+    )
+
+    yield
+
+    while event.type == 'mouse_move':
```

Comment:
It feels like the logic here could be simplified by directly modifying the `layer.translation`, `layer.rotation` or `layer.scale` (i.e., you don't have to modify/create Affine objects).

For example, with translation:
1. store the initial layer translation
2. each mouse move, measure the drag vector from the drag start, add the drag vector to the initial translation to the initial translation and set the `layer.translation`.

Reply:
I guess we need to decide what each of those mean. Currently we always interpreted transforms like this in napari (afaik):

- `Layer.translate`, `rotate`, etc make up the `data2physical` affine, which should take care of things like anisotropicity of sensors, compensating systematic offsets in data collection, etc.
- `Layer.affine` is an entirely separate affine which is defined as `physical2world`, which as I understand is supposed to be for registration.

I think we are in the second case here, and I also wanted keep the logic of the old `InteractionBox`, which acted on the `affine`.

For reference, these are the lines where these transforms are defined:

https://github.com/napari/napari/blob/d1898e24adf49872272075c32645885ffad798f8/napari/layers/base/base.py#L298-L312